src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java

```diff
@@ -70,9 +73,9 @@
 
     public static final String QUERY_ROW_KEY_INDEX = "SELECT column1 FROM row_key_index WHERE key = ? AND column1 >= ? and column1 <= ? ORDER BY column1 ASC LIMIT ?";
 
-    public static final String QUERY_ROW_KEY_SPLIT_INDEX = "SELECT column1 FROM row_key_split_index WHERE metric_name = ? AND tag_name = ? and tag_value IN ? AND column1 >= ? and column1 <= ? LIMIT ?";
+    public static final String QUERY_ROW_TIME_KEY_SPLIT_INDEX = "SELECT column1 FROM row_time_key_split_index WHERE metric_name = ? AND tag_name = ? and tag_value IN ? AND column1 >= ? and column1 <= ? AND time_bucket IN ? LIMIT ?";
```

Comment:
I wouldn't try to use `time_bucket IN` syntax, even if it works.  The reason is that such a query have to touch multiple partitions, but it will be served through a single node, the coordinator for the request.  A better approach is to make multiple requests from the client in parallel and aggregate the results.

Reply:
This shouldn't be a blocker for deploying the write path, though, we can refine this later.